x-pack/legacy/plugins/infra/public/utils/infra_redirect.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { Route, Redirect, RouteProps } from 'react-router-dom';
+
+// This workaround preserves query parameters in the redirect
+// https://github.com/ReactTraining/react-router/issues/5818#issuecomment-379212014
+export const InfraRedirect: React.FunctionComponent<{
+  from: string;
+  to: string;
+  exact?: boolean;
+}> = ({ from, to, exact }) => (
+  <Route
+    path={from}
+    component={({ location }: RouteProps) =>
```

Comment:
If I read the react-router docs correctly using `render` instead of `component` could save us a component hierarchy level.